src/locutus/model/terminology.py

```diff
@@ -649,6 +652,131 @@ def replace_preferred_terminology(self, editor, preferred_terminology):
             print(f"An error occurred while adding preferred terminology: {e}")
             raise
 
+    def get_user_input(self, resource_type, collection_type, id, code, type):
+            """
+            Retrieves user input for the identified Resource/id/collection/code/type.
+            Does not filter down by editor.
+            Returns only one type of user_input.
+
+            Args:
+            resource_type (str): The resource collection (e.g., "Terminology").
+            collection_type (str): The subcollection (e.g., "user_input")
+            id (str): The document ID.
+            code (str): The target document (mapping) identifier.
+            type (str): The type of input to retrieve 
+                (e.g., "mapping_conversations" or "mapping_votes").
+
+            Returns:
+                dict: Serialized user input data based on the type specified.
+            
+            Example output for TerminologyUserInput type mapping_votes
+            {
+                "Terminology": "tm--2VjOxekLP8m28EPRqk95",
+                "code": "TEST_0001",
+                "user_id": [
+                    {
+                        "user_id": user
+                        "up": "date"
+                    },
+                ]
+            }
+            """
+            try:
+                doc_ref = persistence().collection(resource_type).document(id) \
+                    .collection(collection_type).document(code)
+
+                doc_snapshot = doc_ref.get()
+
+                if doc_snapshot.exists:
+                    existing_data = doc_snapshot.to_dict()
+
+                    # Handle different input types
+                    if type == "mapping_conversations":
+                        schema = UserInput.MappingConversations._Schema()
+                        input_data = existing_data.get("mapping_conversations", [])
+                    elif type == "mapping_votes":
+                        schema = UserInput.MappingVotes._Schema()
+                        input_data = existing_data.get("mapping_votes", {})
+                    else:
+                        return {"message": "Invalid input type specified."}, 400
+
+                    # Serialize the data using the appropriate schema
+                    serialized_data = schema.dump({type: input_data})
+
+                    return {"Terminology": id,
+                            "code": code,
+                            type: serialized_data[type]}
+
+                else:
+                    # Return an empty response if no input exists
+                    return {"Terminology": id, "code": code, "message": "No user input for this mapping."}
+            
+            except Exception as e:
+                return (f"An error occurred while retrieving user input for {id} {resource_type} - {code}: {e}"), 500
+
+
+    
+    def create_or_replace_user_input(self, resource_type, collection_type, id, code, type, user_input):

```

Comment:
you could move this function into the UserInput class since it would be the same for both of the children classes. Then whichever one is instantiated will still have the same function available.